libsolidity/codegen/CompilerContext.cpp

```diff
@@ -138,18 +136,41 @@ bool CompilerContext::appendCallback(evmasm::AssemblyItem const& _i) {
 	m_disable_rewrite = true;
 
 	auto callYUL = R"(
-		mstore(add(callBytes, 4), addr)
+		// store _gasLimit
+		mstore(add(callBytes, 0x04), in_gas)
+		// store _address
+		mstore(add(callBytes, 0x24), addr)
+		// store abi byte memory offset bullshit
+		mstore(add(callBytes, 0x44), 0x60)
+		// store bytes memory _calldata.length
+		mstore(add(callBytes, 0x64), argsLength)
+		// store bytes memory _calldata raw data
+		let rawCallBytes := add(callBytes, 0x84)
 		for { let ptr := 0 } lt(ptr, argsLength) { ptr := add(ptr, 0x20) } {
-			mstore(add(add(callBytes, 0x24), ptr), mload(add(argsOffset, ptr)))
+			mstore(add(rawCallBytes, ptr), mload(add(argsOffset, ptr)))
 		}
-		let success := kall(callBytes, add(0x24, argsLength), retOffset, retLength)
-		if eq(success, 0) {
-			let result := mload(0x40)
-			returndatacopy(result, 0, returndatasize())
-			revert(result, returndatasize())
+		// we will return to where the calldata ends.  Overpadded to ensure that we do sufficient trailing zeros for ABI encoding
+		let endOfArgs := add(callBytes, add(0x104, argsLength))
```

Comment:
can we change this 0x104 -> 0x20? This is just to account for argsLength not ending directly at the end of a word right?